tests/integration/backup/test_v3_backup.py

```diff
@@ -426,3 +425,223 @@ def test_can_inspect_v1(self) -> None:
+            check=True,
+        )
+        metadata_path = tmp_path / f"topic-{new_topic.name}" / f"{new_topic.name}.metadata"
+
+        cp = subprocess.run(
+            [
+                "karapace_schema_backup",
+                "verify",
+                f"--location={metadata_path!s}",
+                "--level=file",
+            ],
+            capture_output=True,
+            check=False,
+            env=no_color_env(),
+        )
+
+        assert cp.returncode == 0
+        assert cp.stderr == b""
+        assert f"Integrity of {new_topic.name}:0.data is intact.".encode() in cp.stdout
+        assert b"Verified 1 data files in backup OK." in cp.stdout
+
+    def test_can_verify_record_integrity_from_large_topic(
+        self,
+        tmp_path: Path,
+        new_topic: NewTopic,
+        producer: KafkaProducer,
+        config_file: Path,
+    ) -> None:
+        # Populate the test topic.
+        for _ in range(100):
+            producer.send(
+                new_topic.name,
+                key=1000 * b"a",
+                value=1000 * b"b",
+                partition=0,
+            ).add_errback(_raise)
+        producer.flush()
+
+        # Execute backup creation.
+        subprocess.run(
+            [
+                "karapace_schema_backup",
+                "get",
+                "--use-format-v3",
+                f"--config={config_file!s}",
+                f"--topic={new_topic.name}",
+                f"--location={tmp_path!s}",
+            ],
+            capture_output=True,
+            check=True,
+        )
+        metadata_path = tmp_path / f"topic-{new_topic.name}" / f"{new_topic.name}.metadata"
+
+        cp = subprocess.run(
+            [
+                "karapace_schema_backup",
+                "verify",
+                f"--location={metadata_path}",
+                "--level=record",
+            ],
+            capture_output=True,
+            check=False,
+            env=no_color_env(),
+        )
+
+        assert cp.returncode == 0
+        assert cp.stderr == b""
+        assert f"Integrity of {new_topic.name}:0.data is intact.".encode() in cp.stdout
+        assert b"Verified 1 data files in backup OK." in cp.stdout
+
+    def test_can_refute_file_integrity(self) -> None:
+        metadata_path = Path(__file__).parent.parent.resolve() / "test_data" / "backup_v3_corrupt" / "2db42756.metadata"
+
+        cp = subprocess.run(
+            [
+                "karapace_schema_backup",
+                "verify",
+                f"--location={metadata_path!s}",
+                "--level=file",
+            ],
+            capture_output=True,
+            check=False,
+            env=no_color_env(),
+        )
+
+        assert cp.returncode == 1
+        assert cp.stderr == b""
+        assert b"Integrity of 2db42756:0.data is not intact!" in cp.stdout
+        assert b"Found checksum mismatches." in cp.stdout
+
+    def test_can_refute_record_integrity(self) -> None:
```

Comment:
The backup file can be  invalid also when it has too few or too many records.
Also the checksum can mismatch on the record batch or the last full checksum.

Maybe the test files could be named to reflect the how the file is broken?

Reply:
The filename is now descriptive of its corruption:

```
renamed:    tests/integration/test_data/backup_v3_corrupt/2db42756.metadata -> tests/integration/test_data/backup_v3_corrupt_last_record_bit_flipped_no_checkpoints/2db42756.metadata
renamed:    tests/integration/test_data/backup_v3_corrupt/2db42756:0.data -> tests/integration/test_data/backup_v3_corrupt_last_record_bit_flipped_no_checkpoints/2db42756:0.data
```